tool/tctl/common/device/resource.go

```diff
@@ -39,16 +39,26 @@ type Resource struct {
 // This spec is intended to closely mirror [devicepb.Device] but swaps some data around
 // to get a UX that matches with our other resource types.
 type ResourceSpec struct {
-	// OsType is represented as a string here for user-friendly manipulation.
-	OsType      string                      `json:"os_type"`
+	// OSType is represented as a string here for user-friendly manipulation.
+	OSType      string                      `json:"os_type"`
 	AssetTag    string                      `json:"asset_tag"`
 	CreateTime  time.Time                   `json:"create_time,omitempty"`
 	UpdateTime  time.Time                   `json:"update_time,omitempty"`
 	EnrollToken *devicepb.DeviceEnrollToken `json:"enroll_token,omitempty"`
 	// EnrollStatus is represented as a string here for user-friendly manipulation.
-	EnrollStatus  string                          `json:"enroll_status"`
-	Credential    *devicepb.DeviceCredential      `json:"credential,omitempty"`
-	CollectedData []*devicepb.DeviceCollectedData `json:"collected_data,omitempty"`
+	EnrollStatus  string                     `json:"enroll_status"`
+	Credential    *devicepb.DeviceCredential `json:"credential,omitempty"`
+	CollectedData []CollectedData            `json:"collected_data,omitempty"`
+}
+
+// CollectedData mirrors [devicepb.DeviceCollectedData] but with a different
+// timestamp type to achieve consistent serialization output.
+type CollectedData struct {
+	CollectTime time.Time `json:"collect_time"`
+	RecordTime  time.Time `json:"record_time"`
+	// OSType is represented as a string here for user-friendly manipulation.
+	OSType       string `json:"os_type"`
+	SerialNumber string `json:"serial_number"`
```

Comment:
```suggestion
	SerialNumber string `json:"serial_number,omitempty"`
```

May be absent on Windows/Linux, depending on what we decide for those.